docs/checks/docs_link_checker.py

```diff
@@ -284,6 +288,10 @@ def _check_link(self, match: re.Match, file: str) -> Optional[LinkReport]:
                     result = self._check_absolute_link(
                         link, file, match.group("path"), match.group("version")
                     )
+                elif match := self._absolute_file_pattern.match(link):  # type: ignore[assignment]
```

Comment:
nit: You don't need to set `match` in this `elif` since you don't use it. I do think using this walrus pattern elsewhere in this crazy conditional instead of would make things clearer though since that would prevent us from repeatedly shadowing it.